app/src/main/java/com/cupcake/jobsfinder/ui/posts/PostsViewModel.kt

```diff
@@ -0,0 +1,59 @@
+package com.cupcake.jobsfinder.ui.posts
+
+import androidx.lifecycle.viewModelScope
+import com.cupcake.jobsfinder.domain.model.Post
+import com.cupcake.jobsfinder.domain.usecase.GetPostsUseCase
+import com.cupcake.jobsfinder.ui.base.BaseViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.update
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import java.lang.Exception
+import javax.inject.Inject
+
+@HiltViewModel
+class PostsViewModel @Inject constructor(
+    private val getPostsUseCase: GetPostsUseCase
+): BaseViewModel() {
+
+    private val _uiState = MutableStateFlow(PostsUIState())
+    val uiState: StateFlow<PostsUIState> = _uiState
+    init {
+        onGetPosts()
+    }
+
+    private fun onGetPosts(){
+        _uiState.update { it.copy(isLoading = true) }
+
+        viewModelScope.launch(Dispatchers.IO) {
```

Comment:
if the result showing on ui u need to use main thread

Reply:
thanks mustafa, but I don't directly observe the onGetPosts function. I observe the uiState in the UI layer. However, when observing the uiState, I switch the thread to the main thread in the UI layer